doc/cephfs/purge-queue.rst

```diff
@@ -0,0 +1,64 @@
+============
+Purge Queue
+============
+
+MDS maintains a data structure known as **Purge Queue** which is responsible
+for managing and executing the sequentially deletion of files.
+There is a Purge queue for every MDS rank. Purge queues consist of purge items
+which contain nominal information from the inodes such as size and the layout
+(i.e. all other un-needed metadata information is discarded making it
+independent of all metadata structures).
```

Comment:
Are "purge items" just objects marked for deletion?

Reply:
@joscollin @zdover23 

purge items here are `PurgeItem`s which is a data structure representing journal entries; when dirs or files are deleted, they are added to PurgeQueue in form of purge items from the journal. Technically a purge item can be called a descriptor of the work associated with purging a dir/file. At the end they are all objects, yes. In simpler words we can say they are files (since for every dir purged, it is iterated for all dirfrags which basically corresponds to a file). I was a bit sceptical to include such technical jargons in a user-facing doc. Suggestions welcome.

Reply:
I think the level of technical information that you've provided in this response is exactly the kind of thing that we want our upstream contributors to understand. I'll work this information into the document. Thanks for it.

Reply:
I will include this information in a near-future PR dedicated solely to the explanation of ``Purge Item``s.